code/obj/machinery/drone_beacon.dm

```diff
@@ -0,0 +1,82 @@
+#define PLAYER_SEEK_RANGE 15
+#define DRONE_SEEK_RANGE 25
+#define DRONE_SUMMON_RANGE 15
+
+/obj/machinery/drone_beacon
+	name = "Mysterious Beacon"
+	desc = "Some strange transmitter."
+	icon = 'icons/obj/ship.dmi'
+	icon_state = "beacon_synd"
+	anchored = TRUE
+	density = TRUE
+	processing_tier = PROCESSING_SIXTEENTH
+	_max_health = 500
+	_health = 500
+
+	New()
+		..()
+		src.AddComponent(/datum/component/obj_projectile_damage)
+		RegisterSignal(src, COMSIG_ATOM_HITBY_PROJ, .proc/hitby_proj)
+
+	ex_act(severity)
+		summon_drones()
+		..()
+
+	attackby(obj/item/I, mob/user)
+		summon_drones()
+		..()
+
+	process()
+		var/list/target_list = list()
+		for(var/mob/living/M in mobs)
+			if(isdead(M))
+				continue
+			if(istype(M.loc, /obj/machinery/vehicle) || !IN_RANGE(src, M, PLAYER_SEEK_RANGE))
+				continue
+			target_list += M
```

Comment:
Yeah as I pinged you in Discord (too late probably) it actually turns out that `view` is the one proc that **is** optimized for direct /atom subtypes so you can probably revert this part actually. Sorry! 😰

Reply:
fixed!